lib/SIL/SILInstructions.cpp

```diff
@@ -1715,6 +1715,27 @@ ObjCMethodInst::create(SILDebugLocation DebugLoc, SILValue Operand,
                                        Member, Ty);
 }
 
+CXXVirtualMethodInst::CXXVirtualMethodInst(SILDebugLocation DebugLoc,
+                                           SILValue Operand, SILDeclRef Member,
+                                           SILType Ty)
+    : UnaryInstructionBase(DebugLoc, Operand),
+      MultipleValueInstructionTrailingObjects(
+          this, {Ty, Operand->getType()},
+          {ValueOwnershipKind::Unowned, ValueOwnershipKind::Any}),
```

Comment:
This seems suspect to me; a function pointer should be trivial (i.e. ValueOwnershipKind::Any). @gottesmm, thoughts?

Reply:
This is the wrong semantics. The right semantics IMO are that this should be either forwarding or guaranteed.

Reply:
I think that the adjusted this pointer needs to be any because I intend to call ManagedValue::forBorrowedRValue() to get back to being a managed value for the adjusted this ptr.

The managed value code will assert like so.
```c++
assert(value.getOwnershipKind() == ValueOwnershipKind::Any &&
           "Addresses always have trivial ownership");
```
How would I express forwarding as a relationship?